numpyro/distributions/continuous.py

```diff
@@ -490,6 +490,51 @@ def icdf(self, q):
         return self.loc - self.scale * jnp.log(-jnp.log(q))
 
 
+class Gompertz(Distribution):
+    arg_constraints = {
+        "loc": constraints.positive,
+        "shape": constraints.positive,
+    }
+    support = constraints.positive
+    reparametrized_params = ["loc", "shape"]
+
+    def __init__(self, loc=1.0, shape=1.0, validate_args=None):
+        self.loc, self.shape = promote_shapes(loc, shape)
+        # Alias parameters as they appear in this article
+        # https://www.tandfonline.com/doi/abs/10.1080/03461238.2012.687697
+        self.__a, self.__b = self.loc, self.shape
+        batch_shape = lax.broadcast_shapes(jnp.shape(loc), jnp.shape(shape))
+        super(Gompertz, self).__init__(
+            batch_shape=batch_shape, validate_args=validate_args
+        )
+
+    def sample(self, key, sample_shape=()):
+        assert is_prng_key(key)
+        random_shape = sample_shape + self.batch_shape + self.event_shape
+        unifs = random.uniform(key, shape=random_shape)
+        return self.icdf(unifs)
+
+    @validate_sample
+    def log_prob(self, value):
+        # https://www.wolframalpha.com/input/?i=log%28a+e%5E%28-%28a+%28-1+%2B+e%5E%28b+x%29%29%29%2Fb+%2B+b+x%29%29
+        return jnp.log(self.__a) + self.__b * value - self.__a / self.__b * (jnp.exp(self.__b * value) - 1)
```

Comment:
nit: you can store `concentration * value` and use `jnp.expm1` for better numerical stability.

Reply:
Resolved